### PR TITLE
Forbid use of os.path.join, point user to tmt._compat.Path

### PR DESCRIPTION
**Pull request by happz**
Pull Request Checklist

* [x] implement the feature

### CODE REVIEW

File: tests/discover/data/distgit_test_plugin.py [hunk 1]
```diff
@@ -17,14 +16,14 @@ class TestDistGit(DistGitHandler):
     usage_name = "TESTING"
     server = f"http://localhost:{SERVER_PORT}"
 
-    def url_and_name(self, cwd='.'):
-        with open(os.path.join(cwd, MOCK_SOURCES_FILENAME)) as f:
-            data = f.read()
+    def url_and_name(self, cwd=None):
```
Comment (martinhoyer) [inline tests/discover/data/distgit_test_plugin.py:19]: Was there ever a consensus about the use of `https://docs.python.org/3/library/os.html#os.PathLike` or something to annotate 'pathlike'?

**Review by martinhoyer — APPROVED** [review]
Lovely

File: tests/discover/data/distgit_test_plugin.py (hunk 1 above)
Comment (happz) [reply tests/discover/data/distgit_test_plugin.py:19]: Nope, never discussed there, we use `tmt.utils.Path` everywhere.

**Review by therazix — APPROVED** [review]

**Comment by happz** [comment]
Unrelated failure, merging.

**Pull request merged**